onyo/lib/inventory.py

```diff
@@ -725,3 +725,66 @@ def raise_empty_keys(self, asset: dict | UserDict) -> None:
             # Note, that DotNotationWrapper.keys() delivers strings (and has to).
             # Hence, `None` as a key would show up here as 'None'.
             raise ValueError("Keys are not allowed to be empty or None-values.")
+
+    def _parse_operations_record(self, record: list[str]):
+
+        if not record[0].strip() == "--- Inventory Operations ---":
+            raise RuntimeError("Invalid operations record.")
+        parsed_record = {k: [] for k in OPERATIONS_MAPPING.keys()}
+        collecting_key = None
+        for line in record[1:]:
+            line = line.strip()
+            if not line:
+                continue
+            match line:
+                case "New assets:":
+                    collecting_key = "new_assets"
+                case "New directories:":
+                    collecting_key = "new_directories"
+                case "Removed assets:":
+                    collecting_key = "remove_assets"
+                case "Removed directories:":
+                    collecting_key = "remove_directories"
+                case "Moved assets:":
+                    collecting_key = "move_assets"
+                case "Moved directories:":
+                    collecting_key = "move_directories"
+                case "Renamed directories:":
+                    collecting_key = "rename_directories"
+                case "Renamed assets:":
+                    collecting_key = "rename_assets"
+                case "Modified assets:":
+                    collecting_key = "modify_assets"
+                case _:
+                    if not collecting_key or line and not line.startswith("- "):
+                        raise RuntimeError("Invalid operations record.")
+                    cleaned_line = line[2:].split(" -> ")
+                    if len(cleaned_line) > 2:
+                        raise RuntimeError(f"Invalid operations record:{line}")
+                    if len(cleaned_line) == 1:
+                        parsed_record[collecting_key].append(Path(cleaned_line[0]))
+                    else:
+                        parsed_record[collecting_key].append(
+                            (Path(cleaned_line[0]), Path(cleaned_line[1]))
+                        )
+        return parsed_record
+
+    def get_history(self, path: Path | None = None, n: int | None = None):
+        # TODO: This isn't quite right yet. operations records are defined in Inventory.
```

Comment:
I had a nice paragraph written up about this, but then I realized that I'm not sure if I have a clear understanding of the distinction between OnyoRepo and Inventory.

I see your desire to separate the layers, and not hop directly from Inventory to GitRepo. But what I don't have an answer for yet (and I hope you can help explain), is what is an Onyo Repo other than an Inventory?

The question that makes me ask this is: should OnyoRepo or Inventory be the one to parse the commit log dict, to extract and add info about the Inventory Operations?

Reply:
The idea would be, that from `Inventory`'s POV, `OnyoRepo` is the storage backend. If we were to add a DB-version of this, `Inventory` would refer to a `backend` rather than a `repo` and nothing else changes in `Inventory` or any of the command implementations. So, `OnyoRepo` implements what's specific about an onyo repository, utilizing the generic `GitRepo`. But it's still aware of filesystem and uses paths in the repo. Whereas `Inventory` is supposed to be more abstracted. It knows assets and inventory dirs (which happen to have a path-like labe ATM, but soon will be `Item`s). It doesn't care about how this is stored.

Reply:
So the comment suggests that:
- there may be an `Inventory.get_history(item: Item)` that refers to an `OnyoRepo.get_history(path: Path)`.
- make `Inventory.commit()` not create the commit message yet, but pass the operations record separately to `OnyoRepo` and have only this layer decide, that this is going to be part of the commit message, so that the definition of how it's stored and decoded is at the same layer.

However, it's not super urgent/important. This entire distinction between `Inventory` and `OnyoRepo` helps to have a clear conceptualization of what premises a given piece of code operates on, but it probably won't be perfectly separated w/o an actual second backend anyway.